java-frontend/src/test/java/org/sonar/java/model/JMethodSymbolTest.java

```diff
@@ -45,6 +48,69 @@ void test() {
       .containsOnly(cu.sema.type(Objects.requireNonNull(cu.sema.resolveType("java.lang.Exception"))));
   }
 
+  @Test
+  void testSingleInheritance() {
+    JavaTree.CompilationUnitTreeImpl cu = test("interface A { void a(); }  class C implements A { public void a() { } }");
```

Comment:
The test cases could be slightly rewritten to ease the readiness, for instance having one class/interface by line.